src/agent/communicator/src/http_socket.hpp

```diff
@@ -43,6 +45,56 @@ namespace http_client
             boost::beast::http::read(m_socket, buffer, res);
         }
 
+        void ReadToFile(boost::beast::http::response_parser<boost::beast::http::dynamic_body>& res,
```

Comment:
This implementation and the one in `HttpsSocket` are almost identical, how can we avoid this code duplication?

Reply:
Done!